framework/CodeInterfaces/MELCOR/MELCORdata.py

```diff
@@ -47,108 +55,20 @@ def __init__(self,filen):
-            break
-          match = re.match(regex, line)
-          if match is not None:
-            functionValues[match.groupdict()["name"]] = match.groupdict()["value"]
-      elif start == -2:
-        for lineNumber, line in enumerate(listOfLines):
-          fcnName = re.match(timeOneRegex_name, line)
-          if fcnName is not None:
-            fcnValue = re.match(timeOneRegex_value, listOfLines[lineNumber+1])
-            if fcnValue is not None:
-              functionValues[fcnName.groupdict()["name"]] = fcnValue.groupdict()["value"]
-      functionValuesForEachTime[time] = functionValues
-    return functionValuesForEachTime
-
-  def returnVolumeHybro(self,timeBlock):
-    """
-      CONTROL VOLUME HYDRODYNAMICS EDIT
-      @ In, timeBlock, dict, {"time":[lines Of Output for that time]}
-    """
-    volForEachTime = {}
-    for time,listOfLines in timeBlock.items():
-      results = {}
-      for cnt, line in enumerate(listOfLines):
-        if line.strip().startswith("VOLUME"):
-          headers  = line.strip().split()[1:len(line.strip().split())-1]
-          for lineLine in listOfLines[cnt + 2:]:
-            if len(lineLine.strip()) < 1:
-              break
-            valueSplit   = lineLine.strip().split()
-            volumeNumber = lineLine.strip().split()[0]
-            if not volumeNumber.isdigit():
-              break
-            valueSplit = valueSplit[1:len(valueSplit)]
-            for paramCnt,header in enumerate(headers):
-              parameter = "volume_"+str(volumeNumber)+"_"+header.strip()
-              try:
-                testFloat = float(valueSplit[paramCnt])
-                results[parameter] = valueSplit[paramCnt]
-              except ValueError:
-                # in this way, the "strings" are not placed in the resulting csv
-                pass
-      volForEachTime[time] = copy.deepcopy(results)
-    return volForEachTime
-
-  def writeCsv(self,filen):
+  def writeCsv(self,filen,filen2):
     """
       Output the parsed results into a CSV file
       @ In, filen, str, the file name of the CSV file
       @ Out, None
     """
     IOcsvfile=open(filen,'w+')
-    getHeaders = self.timeParams.values()[0].keys()
-    CFHeaders = self.functions.values()[0].keys()
-    header = ','.join(getHeaders + CFHeaders)
-    header = "time,"+header+"\n"
-    IOcsvfile.write(header)
-    for time in self.timeParams.keys():
-      stringToWrite = str(time)
-      for value in self.timeParams[time].values():
-        stringToWrite+=","+str(value)
-      for value in self.functions[time].values():
-        stringToWrite+=","+str(value)
-      stringToWrite+="\n"
-      IOcsvfile.write(stringToWrite)
-    IOcsvfile.close()
+    file_dir = filen2
+    Var_srch=['CVH-P_1','CVH-P_132','CVH-P_136','CVH-P_137','CVH-P_138','CVH-PPART.3_132','CVH-PPART.4_132','CVH-PPART.5_132','CVH-PPART.6_132','CVH-PPART.7_132','CVH-PPART.8_132','CVH-PPART.9_132','CVH-TLIQ_136','CVH-TVAP_136','CVH-TVAP_137', \
```

Comment:
The code has been modified. Now the user, have to declare the variables to use in the postprocess directly in the MELCOR input deck. The user have to write (everywhere in the MELCOR input file) a string beginning with "**$RAV_VAR**%" followed by variables name separated by comma. Here an example:
*
**$RAV_VAR**%CVH-P_8, CVH-MASS.1_8, CVH-H_8, CVH-VOID-T_8
**$RAV_VAR**%CVH-TVAP_8, CVH-TLIQ_8, CVH-VOID-P_8,
**$RAV_VAR**% ....,....,....
*

Reply:
Matteo,
isn't better to list the variables in the XML input file directly? Something like:
```
	<Code name="MELGEN" subType="Melcor">
	  <executable>MELCOR_2.2/bin/melcor</executable>
	  <preexec>MELCOR_2.2/bin/melgen</preexec>
          <variables>
             CVH-P_8, CVH-MASS.1_8, CVH-H_8, CVH-VOID-T_8,
             CVH-TVAP_8, CVH-TLIQ_8, CVH-VOID-P_8
          </variables>
    </Code>
```

and then just implement the method ```_readMoreXML```:
```
  def _readMoreXML(self,xmlNode):
    """
      Function to read the portion of the xml input that belongs to this specialized class and initialize
      some members based on inputs. This can be overloaded in specialize code interface in order to
      read specific flags.
      Only one option is possible. You can choose here, if multi-deck mode is activated, from which deck you want to load the results
      @ In, xmlNode, xml.etree.ElementTree.Element, Xml element node
      @ Out, None.
    """
    self.variables = []
    varNode = xmlNode.find("variables")
    if varNode is None:
      raise IOError("Melcor variables not found (o il messaggio che ti pare a te)")
    self.variables = [var.strip() for var in varNode.text.split(",")]

```

Reply:
Yes, no problem.

Reply:
Two nodes have to be declared in the xml input file. The first to declare the name of the MELCOR plot file, the latter to declare the variables.

<CodePlotFile>MELCOR_PLT_FILE.PTF</CodePlotFile>
<variables>CVH-P_8,CVH-MASS.1_8,CVH-H_8,CVH-VOID-T_8,
CVH-TVAP_8,CVH-TLIQ_8,CVH-VOID-P_8</variables>